bingo/EA/DeterministicCrowding.py

```diff
@@ -0,0 +1,48 @@
+"""The "Deterministic Crowding" evolutionary algorithm
+
+This module defines the basis of the "deterministic crowding"
+evolutionary algorithm in bingo analyses. The next generation
+is selected by pairing parents with their offspring and
+advancing the most fit of the two.
+"""
+from bingo.Base.EvolutionaryAlgorithm import EvolutionaryAlgorithm
+from bingo.EA.VarAnd import VarAnd
+from bingo.EA.DeterministicCrowdingSelection import DeterministicCrowdingSelection
+
+class DeterministicCrowdingEA(EvolutionaryAlgorithm):
+    """The algorithm used to perform generational steps.
+
+    A class for the "mu comma lambda" evolutionary algorithm in bingo.
+
+    Attributes
+    ----------
+    evaluation : Evaluation
+                 Evaluation instance to perform evaluation on a population
+    selection : DeterministicCrowdingSelection
+                Performs selection on a population via deterministic crowding
+    variation : VarAnd
+                Performs VarAnd variation on a population
+    """
+    def __init__(self, evaluation, crossover, mutation, crossover_probability,
+                 mutation_probability):
+        self._evaluation = evaluation
+        self._selection = DeterministicCrowdingSelection()
+        self._variation = VarAnd(crossover, mutation, crossover_probability,
+                                 mutation_probability)
+
+    def generational_step(self, population):
```

Comment:
is this function the same as the one in mu + lambda?

if so, lets have deterministic crowding extend that class, that way we don't have to rewrite the function

Reply:
never mind, I just realized the difference, the number of offspring is fixed here.

ignore my original comment here, this is good as-is